### PR TITLE
Backend TensorFlow supports forward-mode automatic differentiation

### PR DESCRIPTION
**Pull request by ZongrenZou**

### CODE REVIEW

File: deepxde/data/pde.py [hunk 1]
```diff
@@ -128,8 +128,14 @@ def __init__(
...
     def losses(self, targets, outputs, loss_fn, inputs, model, aux=None):
-        if backend_name in ["tensorflow.compat.v1", "tensorflow", "paddle"]:
+        if backend_name in ["tensorflow.compat.v1", "paddle"]:
             outputs_pde = outputs
+        elif backend_name == "tensorflow":
+            if config.autodiff == "reverse":
+                outputs_pde = outputs
+            elif config.autodiff == "forward":
+                # forward-mode AD in TensorFlow requires functions
+                outputs_pde = (outputs, aux[0])
```
Comment (lululxvi) [inline deepxde/data/pde.py:138]:
```
elif backend_name in ["tensorflow", "pytorch"]:
    if config.autodiff == "reverse":
        outputs_pde = outputs
    elif config.autodiff == "forward":
        # Forward-mode AD requires functions
        outputs_pde = (outputs, aux[0])
```

File: deepxde/gradients/gradients_forward.py [hunk 2]
```diff
@@ -3,7 +3,7 @@
 __all__ = ["hessian", "jacobian"]
 
 from .jacobian import Jacobian, Jacobians
-from ..backend import backend_name, jax, torch
+from ..backend import backend_name, jax, torch, tf
```
Comment (lululxvi) [inline deepxde/gradients/gradients_forward.py:6]:
```
from ..backend import backend_name, jax, tf, torch
```

File: deepxde/gradients/jacobian.py [hunk 3]
```diff
@@ -19,8 +19,16 @@ def __init__(self, ys, xs):
         self.ys = ys
         self.xs = xs
 
-        if backend_name in ["tensorflow.compat.v1", "tensorflow", "paddle"]:
+        if backend_name in ["tensorflow.compat.v1", "paddle"]:
             self.dim_y = ys.shape[1]
+        elif backend_name == "tensorflow":
```
Comment (lululxvi) [inline deepxde/gradients/jacobian.py:24]: Same here. Merge tf and torch.

Comment (ZongrenZou) [reply deepxde/gradients/jacobian.py]: Done.

File: deepxde/gradients/gradients_forward.py (hunk 2 above)
Comment (ZongrenZou) [reply deepxde/gradients/gradients_forward.py]: Done.

File: deepxde/data/pde.py (hunk 1 above)
Comment (ZongrenZou) [reply deepxde/data/pde.py]: Done.

**Pull request merged**